oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java

```diff
@@ -66,97 +68,190 @@ public IndexPrinter(IndexInfoService indexInfoService, AsyncIndexInfoService asy
                 if (stats != null) {
-                        pw.printf("    Status              : %s%n", stats.getStatus());
-                        pw.printf("    Failing             : %s%n", stats.isFailing());
-                        pw.printf("    Paused              : %s%n", stats.isPaused());
+                    keyValue("    Status              ", stats.getStatus(), pw, json, format);
+                    keyValue("    Failing             ", stats.isFailing(), pw, json, format);
+                    keyValue("    Paused              ", stats.isPaused(), pw, json, format);
                     if (stats.isFailing()) {
-                        pw.printf("    Failing since       : %s%n", stats.getFailingSince());
-                        pw.printf("    Latest error        : %s%n", stats.getLatestError());
+                        keyValue("    Failing since       ", stats.getFailingSince(), pw, json, format);
+                        keyValue("    Latest error        ", stats.getLatestError(), pw, json, format);
                     }
                 }
-                pw.println();
+                printWithNewLine(pw, "", format);
             }
+            endJsonObject(json);
         }
+        endJsonObject(json);
     }
 
-    private static void printTitle(PrintWriter pw, String title) {
-        pw.println(title);
-        pw.println(Strings.repeat("=", title.length()));
+    private static void printTitle(PrintWriter pw, String title, Format format) {
+        if (format == Format.TEXT) {
+            pw.println(title);
+            pw.println(Strings.repeat("=", title.length()));
+            pw.println();
+        }
     }
 
-    private void printIndexInfo(PrintWriter pw) {
+    private static void printWithNewLine(PrintWriter pw, String printLine, Format format) {
+        if (format == Format.TEXT) {
+            pw.println(printLine);
+        }
+    }
+
+    private void printIndexInfo(PrintWriter pw, JsopBuilder json, Format format) {
         ListMultimap<String, IndexInfo> infos = ArrayListMultimap.create();
         for (IndexInfo info : indexInfoService.getAllIndexInfo()) {
             infos.put(info.getType(), info);
         }
 
-        pw.printf("Total number of indexes : %d%n", infos.size());
+        keyValue("Total number of indexes ", infos.size(), pw, json, format);
+
         for (String type : infos.keySet()){
             List<IndexInfo> typedInfo = infos.get(type);
             String title = String.format("%s(%d)", type, typedInfo.size());
-            printTitle(pw, title);
-            pw.println();
+            printTitle(pw, title, format);
+            addJsonKey(json, type);
+            startJsonObject(json);
             for (IndexInfo info : typedInfo){
-                printIndexInfo(pw, info);
+                printIndexInfo(pw, json, info, format);
+                if (format == Format.TEXT) {
+                    if (info.hasIndexDefinitionChangedWithoutReindexing()) {
```

Comment:
The json output could also have this info I assume, right?

Reply:
yes - I was thinking that this diff was also in json, and might cause formatting issues - but it's not in json. I have modified for the json stats also to include this